test.py

```diff
@@ -11,6 +11,7 @@
 import re, sys, unittest
 import os.path
 import torch
+import gc
 
 class TestBenchmark(TestCase):
     def setUp(self):
```

Comment:
can you add gc.collect() before this call to memory_allocated()?

this is because we are doing an assert equal rather than an assert less-equal.  It is possible that there is some memory allocated here which simply hasn't been collected yet but is collectible.  I observed on case in debugger where assertion failed with self.memory > memory.